docs/src/main/asciidoc/se/injection.adoc

```diff
@@ -0,0 +1,677 @@
+
+In this example below, we create event producer, which fires event only to observers qualified with name "id".
+[source,java]
+.Qualified event producer
+----
+include::{sourcedir}/se/inject/EventsExample.java[tag=snippet_6, indent=0]
+----
+
+=== Qualified Event Observers
+
+To consume a qualified event, observer method must be annotated with the correct qualifier(s).
+
+[source,java]
+.Qualified event observer
+----
+include::{sourcedir}/se/inject/EventsExample.java[tag=snippet_7, indent=0]
+----
+
+== Asynchronous Events
+Events can be emitted asynchronously, and event observers can also operate asynchronously.
+
+An executor service for handling asynchronous events can be registered in the service registry by providing a service that implements the `java.util.concurrent.ExecutorService` contract and is named `io.helidon.service.registry.EventManager`.
+
+If no custom executor service is provided, the system defaults to a per-task executor using Virtual Threads, with thread names prefixed as `inject-event-manager-`.
+
+=== Asynchronous Event Producer
+All asynchronous event producers must use the `Event.Emitter.emitAsync(..)` method instead of the synchronous `Event.Emitter.emit(..)`.
+
+The `emitAsync` method returns a `CompletionStage<MyEvent>` instance. When executed, it completes once all event observers have been submitted to the executor service. However, there is no guarantee that any event has been delivered—it may have been sent to anywhere from 0 to n observers (where n represents the number of synchronous observers).
+
+[source,java]
+.Asynchronous Event Producer Example
+----
+include::{sourcedir}/se/inject/EventsExample.java[tag=snippet_4, indent=0]
+----
+
+=== Asynchronous Observer
+Asynchronous observer methods are invoked from separate threads (through the executor service mentioned above), and their results are ignored by the Event Emitter; if there is an exception thrown from the observer method, it is logged with `WARNING` log level into logger named `io.helidon.service.registry.EventManager`.
+
+To declare an asynchronous observer use annotation `Event.AsyncObserver` instead of `Event.Observer`.
+
+[source,java]
+.Asynchronous Event Observer Example
+----
+include::{sourcedir}/se/inject/EventsExample.java[tag=snippet_5, indent=0]
+----
+
+== Programmatic Lookup
+
+There are two primary ways to access services from the service registry:
+
+- *Dependency Injection* – Services are injected automatically at defined injection points.
+- *Programmatic Lookup* – Services are retrieved manually by accessing the service registry directly.
+
+If you want to use programmatic lookup, there are two main ways to access a `ServiceRegistry` instance:
+
+- Create a new ServiceRegistry instance and search for the desired service.
+- Inject the existing ServiceRegistry instance currently used by Helidon and retrieve services from it.
```

Comment:
This should only be used for advanced use cases, as it bypasses a lot of optimizations of the service registry (i.e. build time binding of services to injection points when using the Maven plugin)